src/ocp.erl

```diff
@@ -17,25 +17,60 @@
 %%%-----------------------------------------------------------------------
 -module(ocp).
 
--export([start_span/1,
+-export([start_trace/0,
+         start_trace/1,
+         start_span/1,
          finish_span/0,
-         child_span/1]).
+         context/0,
+         put_attribute/2,
+         put_attributes/1]).
 
 -include("opencensus.hrl").
 
+-define(CONTEXT, current_context).
 -define(KEY, current_span).
 -define(SKEY, current_spans).
 
 %%--------------------------------------------------------------------
 %% @doc
-%% Starts a new trace with span stored in the process dictionary.
+%% Creates a new trace context.
+%% @end
+%%--------------------------------------------------------------------
+-spec start_trace() -> ok.
+start_trace() ->
+    put(?CONTEXT, opencensus:start_trace()),
+    put(?SKEY, []).
+
+-spec start_trace(opencensus:trace_context()) -> ok.
+start_trace(TraceContext) ->
```

Comment:
perhaps `start_trace(TraceContext = #trace_context{}) -> ...` to catch bad arguments as early as possible and not having to deal with a odd error somewhere down the road would be nicer? For those not using dialyzer ;)

Reply:
Ah yes, good point. Otherwise it is just plopping whatever in the pdict to crash later.

Hm, though brings up the question of, should we do everything possible to never crash but instead turn into a noop trace in cases like this?

Reply:
That's a really good question! How about we return `error` instead of `ok` and don't push anything? That would not corrupt the context and allow the caller to decide weather they want to crash or not. 

While I like crashing a tracing library probably should not enforce it, it feels nicer to make the user explicitly crash on bad results them forcing them to catch to prevent crashes.

Reply:
Agreed. Though I'm not as sure about not pushing anything... My thought was:

```
start_trace(TraceContext = #trace_context{}) ->
    put(?CONTEXT, TraceContext),
    put(?SKEY, []);
start_trace(_) ->
    put(?CONTEXT, undefined),
    error.
```

since if for some reason there exists a trace context in the pdict and the calling code is going to use it it would be bad to end up using the old trace. Of course if they check for the `error` result they'd know this, but still might be a good safety mechanism to have a noop in the case of bad data instead of more bad data somewhere.

Reply:
You're right, that makes sense corrupting the current trace would be 'bad' the user expects to have pushed something so it should not be the old trace.

I think we'll see this kind of decisions more often so I opened #8 so we can give ourselves some guidelines :)